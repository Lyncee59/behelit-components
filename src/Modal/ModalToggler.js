import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { FasTimes } from '../Icons'

import { theme } from '../Tools/interpolation'

const TogglerIcon = styled(FasTimes)`
  fill: ${theme('modalTogglerColor')};
  &:hover { fill: ${theme('modalTogglerHoverColor')}; }
`

const ModalToggler = ({ ...props }) => <TogglerIcon selectable size="28px" {...props} />

ModalToggler.propTypes = {
  selectable: PropTypes.bool
}

ModalToggler.defaultProps = {
  selectable: true
}

export default ModalToggler
