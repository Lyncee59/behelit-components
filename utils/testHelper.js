import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/Theme'

export const mountWithTheme = (tree) => mount(<ThemeProvider>{tree}</ThemeProvider>)