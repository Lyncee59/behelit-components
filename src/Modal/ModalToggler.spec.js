import React from 'react'
import { mountWithTheme } from '../../utils/testHelper'

import ModalToggler from './ModalToggler'

describe('ModalToggler', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<ModalToggler />)

    expect(tree.find(ModalToggler)).toMatchSnapshot()
  })
})
