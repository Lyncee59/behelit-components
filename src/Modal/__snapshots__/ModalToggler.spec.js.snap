// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalToggler should render correctly 1`] = `
.c1 {
  fill: #990000;
  cursor: pointer;
}

.c1:hover {
  fill: #800000;
}

.c0 {
  fill: #323232;
}

.c0:hover {
  fill: #990000;
}

<ModalToggler
  selectable={true}
>
  <Styled(FasTimes)
    selectable={true}
    size="28px"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-jgZNlG",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  fill: ",
              [Function],
              ";
  &:hover { fill: ",
              [Function],
              "; }
",
            ],
          },
          "displayName": "Styled(FasTimes)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-jgZNlG",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      forwardedRef={null}
      selectable={true}
      size="28px"
    >
      <FasTimes
        className="c0"
        selectable={true}
        size="28px"
      >
        <Styled(SvgFasTimes)
          className="c0"
          height="28px"
          selectable={1}
          width="28px"
        >
          <StyledComponent
            className="c0"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-hzBeZU",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
    ",
                    "
  fill: ",
                    [Function],
                    ";
  cursor: ",
                    [Function],
                    ";

  &:hover {
    fill: ",
                    [Function],
                    ";
  }
",
                    ";
  ",
                  ],
                },
                "displayName": "Styled(SvgFasTimes)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-hzBeZU",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            forwardedRef={null}
            height="28px"
            selectable={1}
            width="28px"
          >
            <SvgFasTimes
              className="c0 c1"
              height="28px"
              selectable={1}
              width="28px"
            >
              <svg
                className="c0 c1"
                height="28px"
                selectable={1}
                viewBox="0 0 352 512"
                width="28px"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </SvgFasTimes>
          </StyledComponent>
        </Styled(SvgFasTimes)>
      </FasTimes>
    </StyledComponent>
  </Styled(FasTimes)>
</ModalToggler>
`;
